apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -578,7 +594,11 @@ def _get_gbp_details(self, context, **kwargs):
         network = core_plugin.get_network(context, port['network_id'])
 
         def is_port_promiscuous(port):
-            return port['device_owner'] == n_constants.DEVICE_OWNER_DHCP
+            if port['device_owner'] == n_constants.DEVICE_OWNER_DHCP:
+                return True
+            if self.port_security_driver and not port['port_security_enabled']:
```

Comment:
Just wondering why it is necessary to check if the extension driver is present ... isn't it sufficient to do:

if not port.get('port_security_enabled', True):
    return True